controllers/nnf_workflow_controller_container_helpers.go

```diff
@@ -86,9 +91,16 @@ func (c *nnfUserContainer) createMPIJob() error {
 	}
 
 	// MPIJobs have two pod specs: one for the launcher and one for the workers
-	launcher := mpiJob.Spec.MPIReplicaSpecs[mpiv2beta1.MPIReplicaTypeLauncher]
+	launcher, found := mpiJob.Spec.MPIReplicaSpecs[mpiv2beta1.MPIReplicaTypeLauncher]
+	if !found {
+		return fmt.Errorf("%s not found in MPIReplicaSpecs", mpiv2beta1.MPIReplicaTypeLauncher)
```

Comment:
Would this allow an admin to root cause the issue?
Same question on line 102 below.

Reply:
Per Dean's comments, I've removed this checks in favor of the webhook. The admin will get a helpful error message upon NnfContainerProfile creation due to the webhook if the launcher/worker specs are missing.